accumulo-store/src/main/java/gaffer/accumulostore/key/core/AbstractCoreKeyBloomFilterIterator.java

```diff
@@ -14,57 +14,52 @@
  * limitations under the License.
  */
 
-package gaffer.accumulostore.key.core.impl;
+package gaffer.accumulostore.key.core;
 
 import gaffer.accumulostore.key.exception.BloomFilterIteratorException;
 import gaffer.accumulostore.utils.AccumuloStoreConstants;
-import gaffer.accumulostore.utils.ByteArrayEscapeUtils;
 import gaffer.accumulostore.utils.IteratorOptionsBuilder;
+import gaffer.accumulostore.utils.Pair;
 import org.apache.accumulo.core.data.Key;
 import org.apache.accumulo.core.data.Value;
 import org.apache.accumulo.core.iterators.Filter;
 import org.apache.accumulo.core.iterators.IteratorEnvironment;
 import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
 import org.apache.hadoop.util.bloom.BloomFilter;
-import java.io.ByteArrayInputStream;
+
 import java.io.DataInputStream;
+import java.io.ByteArrayInputStream;
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.UnsupportedEncodingException;
-import java.util.Arrays;
 import java.util.Map;
 
 /**
  * The BloomFilterIterator should filter out elements based on their membership
- * of the provided bloomFilter. This implementation may not work as desired
- * depending on your {@link gaffer.accumulostore.key.AccumuloKeyPackage}
- * implementation.
+ * of the provided bloomFilter. An implementation of this class should be provided for every {@link gaffer.accumulostore.key.AccumuloKeyPackage}
+ * implementation which extracts vertices correctly for that key package.
  */
-public class CoreKeyBloomFilterIterator extends Filter {
+public abstract class AbstractCoreKeyBloomFilterIterator extends Filter {
 
-    private BloomFilter filter;
+    protected BloomFilter filter;
 
     @Override
+    @edu.umd.cs.findbugs.annotations.SuppressFBWarnings(
+            value = "UWF_FIELD_NOT_INITIALIZED_IN_CONSTRUCTOR",
+            justification = "Filter initialised in the init method")
     public boolean accept(final Key key, final Value value) {
-        return filter.membershipTest(
-                new org.apache.hadoop.util.bloom.Key(getVertexFromKey(key.getRowData().getBackingArray())));
-    }
-
-    protected byte[] getVertexFromKey(final byte[] key) {
-        int pos = -1;
-        for (int i = 0; i < key.length; ++i) {
-            if (key[i] == ByteArrayEscapeUtils.DELIMITER) {
-                pos = i;
-                break;
-            }
-        }
-        if (pos != -1) {
-            return Arrays.copyOf(key, pos);
+        Pair<byte[]> vertices = getVertices(key.getRowData().getBackingArray());
+        if (vertices.getFirst() == null) {
+            return true;
         } else {
-            return key;
+            boolean inSrc = filter.membershipTest(new org.apache.hadoop.util.bloom.Key(vertices.getFirst()));
+            boolean inDst = filter.membershipTest(new org.apache.hadoop.util.bloom.Key(vertices.getSecond()));
+            return inSrc || inDst;
```

Comment:
It would be slightly more efficient to return straight away if inSrc is true. That way we could skip calculating the inDst value.